server/contexts/ctxerr/ctxerr.go

```diff
@@ -209,14 +220,14 @@ func Unwrap(err error) error {
 
 // MarshalJSON provides a JSON representation of a whole error chain.
 func MarshalJSON(err error) ([]byte, error) {
-	chain := make([]interface{}, 0)
+	chain := make([]fleetErrorJSON, 0)
```

Comment:
I got a bit confused by those similarly-named `FleetError`, `FleetErrorChainJSON`, `fleetErrorJSON` and to some extent `StoredError`. Also the fact that `FleetError` has both `toErrorJSON` and `MarshalJSON` implementations. I'm sure they all have their uses but - assuming we have to keep all of them -, we should at least add a documentation block in the file that explains how they relate to each other and what role they play and make sure we keep everything we can unexported to limit the scope (in particular does `FleetErrorChainJSON` need to be exported? It seems to be used only in a test outside this package, if so we should find a way to test without exporting it).

Reply:
@mna 100% agreed it's confusing :(

I figured a way to clean up most of these confusing types, but I'm wondering about `FleetErrorChainJSON`, what do you think if instead of returning this JSON from `errorstore.Retrieve`:

```json
[
  "count": 3,
  "error": {
    "cause": { "message": "...", "stack": [...] },
    "wraps": [{ "message": "...", "stack": [...] }, { "message": "...", "stack": [...] }],
  }
]
```

we return something like:

```json
[
  "count": 3,
  "error":  [{ "message": "...", "stack": [...] }, { "message": "...", "stack": [...] }],
  ]
]
```

That contains the root error (first element) and the wraps all together? it _might_ also make things less confusing for the consumers of the endpoint.